src/System.Net.Primitives/src/System/Net/NetworkCredential.cs

```diff
@@ -76,14 +85,15 @@ public string UserName
         /// </devdoc>
         public string Password
         {
-            get
-            {
-                return InternalGetPassword();
-            }
-            set
-            {
-                _password = value;
-            }
+            get { return _password; }
+            set { _password = value ?? string.Empty; }
```

Comment:
The only reason that would make sense is because it's passed by reference and not by copy, but wouldn't make sense to have old if we are assigning `_password` to a new value right away and discarding the old value. I think it can be done as @danmosemsft suggests :)